crates/nri-resctrl-plugin/src/lib.rs

```diff
@@ -849,4 +1085,371 @@ mod tests {
+        let gp = std::path::PathBuf::from("/sys/fs/resctrl/pod_u1");
+        fs.add_dir(&gp);
+        fs.add_file(&gp.join("tasks"), "");
+
+        // Define pod and container
+        let pod = nri::api::PodSandbox {
+            id: "sb1".into(),
+            uid: "u1".into(),
+            ..Default::default()
+        };
+        let linux = nri::api::LinuxContainer {
+            cgroups_path: "/cg/x:cri-containerd:c1".into(),
+            ..Default::default()
+        };
+        let container = nri::api::Container {
+            id: "c1".into(),
+            pod_sandbox_id: pod.id.clone(),
+            linux: protobuf::MessageField::some(linux),
+            ..Default::default()
+        };
+        let full_cg = nri::compute_full_cgroup_path(&container, Some(&pod));
+
+        // Initially PIDs unassignable (ESRCH)
+        mock_pid_src.set_pids(full_cg.clone(), vec![101, 102]);
+        fs.set_missing_pid(101);
+        fs.set_missing_pid(102);
+
+        // Run pod + add container → expect counts 1/0
+        let ctx = TtrpcContext {
+            mh: ttrpc::MessageHeader::default(),
+            metadata: std::collections::HashMap::new(),
+            timeout_nano: 5_000,
+        };
+        let state_req = StateChangeEvent {
+            event: Event::RUN_POD_SANDBOX.into(),
+            pod: protobuf::MessageField::some(pod.clone()),
+            container: protobuf::MessageField::none(),
+            special_fields: SpecialFields::default(),
+        };
+        let _ = plugin.state_change(&ctx, state_req).await.unwrap();
+        let create_req = CreateContainerRequest {
+            pod: protobuf::MessageField::some(pod.clone()),
+            container: protobuf::MessageField::some(container.clone()),
+            special_fields: SpecialFields::default(),
+        };
+        let _ = Plugin::create_container(&plugin, &ctx, create_req)
+            .await
+            .unwrap();
+
+        // Drain two events (pod created Exists and container accounted)
+        let _ = timeout(Duration::from_millis(100), rx.recv()).await; // pod exists
+        let _ = timeout(Duration::from_millis(200), rx.recv()).await; // container accounted
+
+        // Make current PIDs assignable by clearing missing flags
+        fs.clear_missing_pid(101);
+        fs.clear_missing_pid(102);
+
+        // Retry just this container → expect transition to Reconciled and one event with counts 1/1
+        let st = plugin.retry_container_reconcile("c1").expect("retry ok");
+        assert_eq!(st, ContainerSyncState::Reconciled);
+        // Validate internal state updated and counts improved
+        {
+            let inner = plugin.state.lock().unwrap();
+            let ps = inner.pods.get("u1").expect("pod state");
+            assert_eq!(ps.total_containers, 1);
+            assert_eq!(ps.reconciled_containers, 1);
+            let cs = inner.containers.get("c1").expect("container");
+            assert_eq!(cs.state, ContainerSyncState::Reconciled);
```

Comment:
get the pids from resctrl and verify they include 101 and 102 to verify that the code actually did the reconciliation